File: src/main/java/dev/akif/cookbook/recipe/RecipeInstructionEntity.java
package dev.akif.cookbook.recipe;

import jakarta.persistence.*;
import lombok.*;

import java.io.Serializable;

@AllArgsConstructor
@Entity(name = "recipe_instruction")
@Getter
@IdClass(RecipeInstructionEntity.RecipeInstructionId.class)
@NoArgsConstructor
@Setter
@Table(name = "recipe_instructions")
@ToString
public class RecipeInstructionEntity {
    @Id
    @Column(name = "recipe_id", nullable = false)
    private Long recipeId;

    @Id
    @Column(name = "sort_order", nullable = false)
    private Integer sortOrder;

    @Column(name = "instruction", nullable = false)
    private String instruction;

    @Data
    @NoArgsConstructor
    public static class RecipeInstructionId implements Serializable {
        private Long recipeId;
        private Integer sortOrder;
    }
}

File: src/main/kotlin/dev/akif/cookbook/recipe/instruction/RecipeInstructionEntity.kt
package dev.akif.cookbook.recipe.instruction

import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Table

@Table(name = "recipe_instructions")
class RecipeInstructionEntity(
    @Id
    var id: Long? = null,
    var recipeId: Long = -1,
    var sortOrder: Int = 0,
    var instruction: String = ""
)
